x-pack/plugins/ml/public/application/memory_usage/memory_tree_map/job_memory_tree_map.tsx

```diff
@@ -0,0 +1,170 @@
+import { LoadingWrapper } from '../../jobs/new_job/pages/components/charts/loading_wrapper';
+import { useFieldFormatter } from '../../contexts/kibana';
+
+import { useRefresh } from '../../routing/use_refresh';
+import { getMemoryItemColor } from '../memory_item_colors';
+
+interface Props {
+  node?: string;
+  type?: MlSavedObjectType;
+  height?: string;
+}
+
+const DEFAULT_CHART_HEIGHT = '400px';
+
+const TYPE_OPTIONS: TreeMapOptions[] = [
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.adLabel', {
+      defaultMessage: 'Anomaly detection jobs',
+    }),
+    objectType: 'anomaly-detector',
+    color: getMemoryItemColor('anomaly-detector'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.dfaLabel', {
+      defaultMessage: 'Data frame analytics jobs',
+    }),
+    objectType: 'data-frame-analytics',
+    color: getMemoryItemColor('data-frame-analytics'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.modelsLabel', {
+      defaultMessage: 'Trained models',
+    }),
+    objectType: 'trained-model',
+    color: getMemoryItemColor('trained-model'),
+  },
+];
+
+type TreeMapOptions = EuiComboBoxOptionOption & { objectType: MlSavedObjectType };
+
+export const JobMemoryTreeMap: FC<Props> = ({ node, type, height }) => {
+  const bytesFormatter = useFieldFormatter(FIELD_FORMAT_IDS.BYTES);
+  const refresh = useRefresh();
+  const chartHeight = height ?? DEFAULT_CHART_HEIGHT;
+
+  const trainedModelsApiService = useTrainedModelsApiService();
+  const [allData, setAllData] = useState<MemoryUsageInfo[]>([]);
+  const [data, setData] = useState<MemoryUsageInfo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState<TreeMapOptions[]>(TYPE_OPTIONS);
+
+  const filterData = useCallback(
+    (dataIn: MemoryUsageInfo[]) => {
+      const labels = selectedOptions.map((o) => o.objectType);
+      return dataIn.filter((d) => labels.includes(d.type));
+    },
+    [selectedOptions]
+  );
+
+  const onTypeChange = useCallback((types: EuiComboBoxOptionOption[]) => {
+    setSelectedOptions(types as TreeMapOptions[]);
+  }, []);
+
+  const loadJobMemorySize = useCallback(async () => {
+    setLoading(true);
+    const resp = await trainedModelsApiService.memoryUsage(type, node);
+    setAllData(resp);
+    setLoading(false);
+  }, [trainedModelsApiService, type, node]);
+
+  useEffect(
+    function redrawOnFilterChange() {
+      setData(filterData(allData));
+    },
+    [selectedOptions, allData, filterData]
+  );
+
+  useEffect(
+    function updateOnTimerRefresh() {
+      loadJobMemorySize();
+    },
+    [loadJobMemorySize, refresh]
+  );
+
+  return (
+    <div
+      style={{ height: chartHeight }}
+      data-test-subj={`mlJobTreeMap ${data.length ? 'withData' : 'empty'}`}
+    >
+      <EuiSpacer size="s" />
+      <LoadingWrapper height={chartHeight} hasData={data.length > 0} loading={loading}>
+        <EuiFlexGroup>
```

Comment:
This could do with some empty state callout / text, such as ' There are no open jobs or trained models matching the search criteria.'

<img width="1343" alt="image" src="https://user-images.githubusercontent.com/7405507/216313502-127c78ff-c989-44bd-9daa-038fd9ddeec2.png">

Reply:
Added empty selection prompt in https://github.com/elastic/kibana/commit/69a48df2413c8fed84c22f098f96d2629b694975

![image](https://user-images.githubusercontent.com/22172091/216581547-ddb5c72e-ccbe-4f66-827f-7ecc538c6393.png)